index.js

```diff
@@ -410,7 +410,10 @@ exports.parseNoPatch = function (code, options) {
       ast = jsParse(code, opts);
     }
   } catch (err) {
-    if (err instanceof SyntaxError) {
+    // Any error with location information should be promoted to a code frame.
+    // Otherwise linter plugins for IDEs lose the positional information for
+    // semantic errors.
+    if (err.loc) {
```

Comment:
Sure, I think this looks good... don't have much context on any possible subtleties here...